frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { match as RMatch } from 'react-router';
+import {
+  navFactory,
+  ResourceSummary,
+  SectionHeading,
+  LoadingBox,
+  StatusBox,
+  FirehoseResult,
+} from '@console/internal/components/utils';
+import { SecretModel } from '@console/internal/models';
+import { ErrorPage404 } from '@console/internal/components/error';
+import { DetailsPage } from '@console/internal/components/factory';
+import { K8sResourceKindReference, K8sResourceKind } from '@console/internal/module/k8s';
+
+const SecretReference: K8sResourceKindReference = 'Secret';
+const HelmReleaseReference = 'HelmRelease';
+
+export interface HelmReleaseDetailsPageProps {
+  match: RMatch<{
+    ns?: string;
+  }>;
+  secret?: FirehoseResult;
+}
+
+export interface HelmReleaseDetailsProps {
+  obj: K8sResourceKind;
+}
+
+export const HelmReleaseDetails: React.FC<HelmReleaseDetailsProps> = ({ obj: resourceDetails }) => {
+  const helmRelease = resourceDetails;
+  helmRelease.metadata.name = _.get(resourceDetails, 'metadata.labels.name');
+
+  return (
+    <div className="co-m-pane__body">
+      <SectionHeading text="Helm Release Overview" />
+      <div className="row">
+        <div className="col-sm-6">
+          <ResourceSummary resource={helmRelease} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const HelmReleaseDetailsPage: React.FC<HelmReleaseDetailsPageProps> = ({ secret, match }) => {
+  const namespace = match.params.ns;
+  if (!secret || (!secret.loaded && _.isEmpty(secret.loadError))) {
+    return <LoadingBox />;
+  }
+  if (secret.loadError) {
+    return <StatusBox loaded={secret.loaded} loadError={secret.loadError} />;
+  }
+  const secretResource = secret.data;
+  if (!_.isEmpty(secretResource)) {
+    return (
+      <DetailsPage
+        match={match}
+        kindObj={SecretModel}
+        name={_.get(secretResource[0], 'metadata.name')}
+        namespace={namespace}
+        breadcrumbsFor={() => [
+          {
+            name: `Helm Releases`,
+            path: `/helm-releases/ns/${namespace}`,
+          },
+          { name: `Helm Release Details`, path: `${match.url}` },
+        ]}
+        title={_.get(secretResource[0], 'metadata.labels.name')}
+        kind={SecretReference}
+        pages={[navFactory.details(HelmReleaseDetails)]}
+        customKind={HelmReleaseReference}
+      />
+    );
+  }
+  return <ErrorPage404 />;
```

Comment:
Why are you returning 404 error page here if no secrets are there? Better handle it gracefully and show some message showing the release doesn't exist. It feel wrong to show error page, seems like the page doesn't exist when this error page is seen.

Reply:
@rohitkrai03 the ErrorPage404 is shown if user manually enters an incorrect release name